lib/galaxy/app.py

```diff
@@ -127,15 +136,25 @@ def __init__( self, **kwargs ):
             self.openid_providers = OpenIDProviders.from_file( self.config.openid_config_file )
         else:
             self.openid_providers = OpenIDProviders()
-        # Start the heartbeat process if configured and available
         from galaxy import auth
         self.auth_manager = auth.AuthManager( self )
+        # Start the heartbeat process if configured and available (wait until
+        # postfork if using uWSGI)
         if self.config.use_heartbeat:
             from galaxy.util import heartbeat
```

Comment:
There are others that are imported this way, seems to be mostly historic? It might've had to do with dependencies.